google-cloud-pubsub/lib/google/cloud/pubsub/async_publisher.rb

```diff
@@ -224,40 +284,91 @@ def run_background
           end
         end
 
-        def publish_batch!
-          return unless @batch
+        def resolve_batch_for_message msg
+          @batches[msg.ordering_key] ||= Batch.new self, msg.ordering_key
```

Comment:
Where do remove the Batch object for ordering keys once we have flushed all of the messages? If we don't, we'll have a slow memory leak if one publishes on some ordering keys and then never publishes on them again.

Reply:
The current implementation does not remove empty batches.

Reply:
Empty batches are now removed on publisher flush.